content/tutorials/permissionless-paymaster/10.index.md

```diff
@@ -0,0 +1,439 @@
+
+```javascript
+    import { Wallet } from "zksync-ethers";
+    const signer = Wallet.createRandom();
+```
+
+## 2. Deposit gas funds and add the signer address
+Navigate to ZKsync Sepolia Testnet Explorer: [0xc1B0E2edC4cCaB51A764D7Dd8121CBf58C4D9E40](https://sepolia.explorer.zksync.io/address/0xc1B0E2edC4cCaB51A764D7Dd8121CBf58C4D9E40#contract)
+Using a **different address** with test funds, call the `depositAndAddSigner()` function with 0.1 ether
+and the signer address as the function arguments.
+![deposit and add a signer](/images/permissionless-paymaster/depositAndAddSigner.png)
+
+::callout{icon="i-heroicons-exclamation-circle"}
+The depositor address is considered a "manager". A manager can deposit/withdraw gas funds and add/remove signers at any given time.
+
+- *A signer can only be linked to one manager at a time.*
+
+- *A manager can be a signer address too. (not recommended)*
+
+::
+
+## 3. Setup a basic frontend project
+
+We're going to use the [React.js web framework](https://react.dev/) to build this app.
+In addition, we're going to use the `zksync-ethers` SDK, `ethers@v5`, and `Next.js`.
+In order to focus on paymaster, we have bootstrapped a [template using `zksync-cli create`](https://github.com/ondefy/basic-frontend-template.git).
+We'll just need to implement the methods to sign and integrate paymaster into the frontend.
+Let's start!
+
+1. Clone the template project and `cd` into the folder.
+
+    ```sh
+    git clone https://github.com/ondefy/basic-frontend-template.git
+    cd ./basic-frontend-template
+    ```
+
+2. Make sure you are on the `main` branch(use `git branch` to check) and spin up the project.
+
+    ```bash
+    yarn
+    yarn dev
+    ```
+
+Navigate to `http://localhost:3000/` in a browser to see the application running.
+
+![frotend](/images/permissionless-paymaster/frontend.png)
+
+This is a simple template that allows users to approve a certain amount of DAI tokens to a spender address on ZKsync Sepolia Testnet.
+
+*Our goal for this tutorial is to sponsor this approval transaction using the paymaster.*
+
+---
+
+## 4. Create a function to sign EIP-712 typed paymaster data
+This paymaster verifies the signature signed by the signer address on the below data.
+On successful validation, it allows sponsorship for the user using manager's deposited gas funds.
+
+```solidity
+ bytes32 messageHash = keccak256(
+ abi.encode(
+ SIGNATURE_TYPEHASH,
+ _from,
+ _to,
+ _expirationTime,
+ _maxNonce,
+ _maxFeePerGas,
+ _gasLimit
+ )
+ );
+```
+
+::callout{icon="i-heroicons-exclamation-triangle"}
+For the sake of this tutorial, we will be creating the signing function in the repo itself at *"src/backend/getSignatureApiCall.tsx"* path.
+<br/><br/>
+**It is highly recommended, that signing should be done in a closed backend server for the safety of signer's private key.**
+::
+
+- Create a `.env` file in the root folder and add signer's private key as `NEXT_PUBLIC_SIGNER_PRIVATE_KEY`. Run `source .env` once done.
```

Comment:
I tried to support `getStaticProps` and other data fetching interfaces. The template would require quite a lot of changes. If it's okay, we can move forward since we have cautioned it in the step.